### PR TITLE
[TS] LPS-198515 Error message not displayed on UI after exceeding the Upload Servlet Request limit

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

Motivation
=======
If the length of a Journal Article's content exceeds the configured Upload Servlet Request limit (in bytes) the creation of the Article fails but no message is displayed on the UI indicating the cause of the error.
Only a warning is logged to the console which may not be readily visible to the end user.
The root cause of the issue is that the POST request initiated [here](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/evaluation.es.js#L135-L167) will fail because of the `UploadException` but it is not handled properly on the backend, in the class `DDMFormContextProviderServlet`. 

Proposed Solution
========
Adding logic that processes the exception in `DDMFormContextProviderServlet` and displays the error message on the front end.

Steps to Verify
========
Please see the linked LPS for the reproduction steps.
https://liferay.atlassian.net/browse/LPS-198515

Thank you and best regards,
István

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/servlet/DDMFormContextProviderServlet.java [hunk 1]
```diff
@@ -205,6 +211,31 @@ private void _prepareThreadLocal(Locale locale)
...
+		throws IOException {
+
+		UploadException uploadException =
+			(UploadException)httpServletRequest.getAttribute(
+				WebKeys.UPLOAD_EXCEPTION);
+
+		if ((uploadException != null) &&
+			(uploadException.isExceededLiferayFileItemSizeLimit() ||
+			 uploadException.isExceededFileSizeLimit() ||
+			 uploadException.isExceededUploadRequestSizeLimit())) {
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/servlet/DDMFormContextProviderServlet.java:226]: @IstvanD please sort this alphabetically

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/servlet/DDMFormContextProviderServlet.java [hunk 2]
```diff
@@ -63,6 +65,10 @@ public void service(
 			HttpServletResponse httpServletResponse)
 		throws IOException, ServletException {
 
+		if (_processExceptions(httpServletRequest, httpServletResponse)) {
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/servlet/DDMFormContextProviderServlet.java:68]: @IstvanD please inline as this method is only used once

```
public void service(
		HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse)
	throws IOException, ServletException {

	UploadException uploadException =
		(UploadException)httpServletRequest.getAttribute(
			WebKeys.UPLOAD_EXCEPTION);

	if ((uploadException != null) &&
		(uploadException.isExceededFileSizeLimit() ||
		 uploadException.isExceededUploadRequestSizeLimit() ||
		 uploadException.isExceededLiferayFileItemSizeLimit())) {

		httpServletResponse.sendError(
			HttpServletResponse.SC_REQUEST_ENTITY_TOO_LARGE,
			_language.get(
				httpServletRequest, "upload-size-limit-has-been-exceeded"));

		return;
	}

	createContext(httpServletRequest, httpServletResponse);

	super.service(httpServletRequest, httpServletResponse);
}
```

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/evaluation.es.js [hunk 3]
```diff
@@ -153,6 +153,10 @@ const doEvaluate = debounce((fieldName, evaluatorContext, callback) => {
 		url: EVALUATOR_URL,
 	})
 		.then((newPages) => {
+			if (newPages.statusCode) {
+				callback(newPages);
+			}
```
Comment (carolmariaabb) [inline modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/evaluation.es.js:158]: @alinedoleron Could you quickly review this change?? Thanks!

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js [hunk 4]
```diff
@@ -143,6 +143,10 @@ export default function _JournalPortlet({
 		publishingLock.unlock();
 		console.error(error);
 
+		if (error.error?.statusCode) {
+			showAlert(error.error.message);
+		}
```
Comment (carolmariaabb) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js:148]: @IstvanD the @liferay-echo team needs to review this change

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by IstvanD** [comment]
Hi @carolmariaabb I made the requested changes and force-pushed them.

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js (hunk 4 above)
Comment (IstvanD) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js:148]: @carolmariaabb I have sent it to the Echo Team as well:
https://github.com/liferay-echo/liferay-portal/pull/14050

**Review by carolmariaabb — APPROVED** [review]
@IstvanD Backend related changes look good to me! 

Waiting the FE review.

**Review by alinedoleron — APPROVED** [review]
LGTM!

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:forms

**Comment by IstvanD** [comment]
Hi @carolmariaabb the Echo Team gave it a green light:
https://github.com/liferay-echo/liferay-portal/pull/14050#issuecomment-1772489533

**Comment by carolmariaabb** [comment]
ci:test:forms

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**